primitives/primproc/bppsendthread.h

```diff
@@ -124,7 +141,10 @@ class BPPSendThread
 
     /* secondary queue size restriction based on byte size */
     volatile uint64_t currentByteSize;
-    uint64_t maxByteSize;
+    static uint64_t maxByteSize;
+    // Used to tell the PriorityThreadPool It should consider additional threads because a 
+    // queue full event has happened and a thread has been blocked.
+    boost::shared_ptr<threadpool::PriorityThreadPool> fProcessorPool;
```

Comment:
Could you elaborate on why do we need smart pointer here?

Reply:
There can be only one processor pool so it must be safe to use plain PriorityThreadPool pointer.

Reply:
It's a shared pointer in PrimitiveServer is why it became a shared pointer in BPPSendThread. Since PrimitiveServer is a singleton whose life, once created, lasts until process termination, and fProcessorPool, is a singleton within it, I don't see why it needs to be a shared pointer.